smithy-aws-traits/src/main/resources/META-INF/smithy/aws.api.smithy

```diff
@@ -214,6 +214,40 @@ structure service {
     endpointPrefix: String
 }
 
+/// Annotates a service as having tagging on 1 or more resources and associated
+/// APIs to perform CRUD operations on those tags
+@trait(selector: "service")
+structure tagEnabled {}
+
+/// Indicates a resource supports CRUD operations for tags. Either through
+/// resource lifecycle or instance operations or tagging operations on the
+/// service.
+@trait(selector: "resource")
+structure taggable {
+    /// The `property` property is a string value that identifies which
+    /// resource property represents tags for the resource.
+    property: String
```

Comment:
I'm on the fence here. The docs are a bit weird here because it's a property of the trait named 'property' but I don't like the redundancy of `resource` for a trait applied only to resource shapes to begin with.

Reply:
Further discussions led to a decision to keep the name 'property' here.